filesmanager/filesmanager.py

```diff
@@ -48,20 +132,339 @@ def student_view(self, context=None):
-    # than one handler, or you may not need any handlers at all.
+    def studio_view(self, context=None):
+        """
+        The edit view of the FilesManagerXBlock in Studio.
+        """
+        if context:
+            pass  # TO-DO: do something based on the context.
+        html = self.resource_string("static/html/filesmanager.html")
+        frag = Fragment(html.format(self=self))
+        frag.add_css(self.resource_string("static/css/filesmanager.css"))
+
+        # Add i18n js
+        statici18n_js_url = self._get_statici18n_js_url()
+        if statici18n_js_url:
+            frag.add_javascript_url(self.runtime.local_resource_url(self, statici18n_js_url))
+
+        frag.add_javascript(self.resource_string("static/js/src/filesmanager.js"))
+        frag.initialize_js('FilesManagerXBlock')
+        return frag
+
+    @XBlock.json_handler
+    def get_directories(self, data, suffix=''):
+        """Get the list of directories.
+
+
+        Returns: the list of directories is stored in the XBlock settings.
+
+        [
+            {
+                "name": "Folder 1",
+                "type": "directory",
+                "path": "Folder 1",
+                "metadata": {
+                    "id": ..,
+                    ...
+                },
+                "children": [
+                    ...
+                ]
+            }
+        ]
+        """
+        return {
+            "status": "success",
+            "content": self.directories,
+        }
+
     @XBlock.json_handler
-    def increment_count(self, data, suffix=''):
+    def clear_directories(self, data, suffix=''):
+        """Clear the list of directories without removing files from course assets.
+
+        This method is intended to be used for testing purposes.
+
+        Returns: an empty list of directories.
         """
-        An example handler, which increments the data.
+        self.directories = []
+        self.incremental_directory_id = 0
+        return {
+            "status": "success",
+            "content": self.directories,
+        }
+
+    @XBlock.json_handler
+    def get_content(self, data, suffix=''):
+        """Get the content of a directory.
+
+        Arguments:
+            data: the path of the directory to be retrieved.
+
+        Returns: the content of the directory if found, an empty sequence otherwise.
         """
-        if suffix:
-            pass  # TO-DO: Use the suffix when storing data.
-        # Just to show data coming in...
-        assert data['hello'] == 'world'
+        path = data.get("path")
+        if not path:
+            return {
+                "status": "error",
+                "message": "Path not found",
+            }
+        content, _, _ = self.get_content_by_path(path)
+        return {
+            "status": "success",
+            "content": content,
+        }
+
+    @XBlock.json_handler
+    def add_directory(self, data, suffix=''):
```

Comment:
I think we should avoid duplicate names at the same level.

Reply:
what do you mean?

Reply:
Prevent adding more than one directory with the same name.

Reply:
There should be a constraint on content name, but I prefer to manage that on the frontend side to avoid adding more complexity to this implementation. @johnvente do you think we can do that? For example, when I create a new folder on my file system with a name that already exists, the new folder is created with `New Folder (1)`. Can we do something like that?

Reply:
I think we shouldn't give that responsibility to the frontend, we cannot assume the frontend will always send us right data

Reply:
Why do you think so? I lean a bit more toward the frontend solution cause I'm worried about the amount of processing we're doing synchronously in the handlers to return an error saying the content cannot be created. But we'll see how it behaves.

Reply:
I made the changes, could you folks check again? Thanks